libraries/outputs.rb

```diff
@@ -24,6 +24,27 @@ def self.configure_outputs(node) # rubocop:disable Metrics/PerceivedComplexity,
         else
           [CernerSplunk.my_cluster(node)]
         end.each do |(cluster, bag)|
+          if bag['indexer_discovery'] == true
+            output_stanzas["indexer_discovery:#{cluster}"] = {}
+            if bag['master_uri'].nil? || bag['master_uri'].empty?
+              Chef::Log.warn "master_uri is missing in the cluster databag: #{cluster}"
+            else
+              output_stanzas["indexer_discovery:#{cluster}"]['master_uri'] = bag['master_uri']
+            end
+            encrypt_password = CernerSplunk::ConfTemplate::Transform.splunk_encrypt node: node
+            pass =
+              if bag['indexer_discovery_settings'] && bag['indexer_discovery_settings']['pass4SymmKey']
+                bag['indexer_discovery_settings']['pass4SymmKey']
+              else
+                'changeme'
+              end
+            output_stanzas["indexer_discovery:#{cluster}"]['pass4SymmKey'] = CernerSplunk::ConfTemplate.compose encrypt_password, CernerSplunk::ConfTemplate::Value.constant(value: pass)
+            output_stanzas["tcpout:#{cluster}"] = {}
+            output_stanzas["tcpout:#{cluster}"]['indexerDiscovery'] = cluster
```

Comment:
Doesn't line 56 below remove this from the output stanzas? Seems like it would need to be updated to `|= {}`.  Is it even valid to list both servers and indexer_discovery in the same tcpout stanza?  We may need some additional error checking to say a given cluster can only configure one of receivers or indexer_discovery. And if indexer discovery is on, then don't log the warning about an incomplete cluster data bag on line 54. Another alternative would be to document that if both are present then the configured receivers take precedence over indexer discovery, though I think I would prefer to error in that case.

Reply:
My plan was that indexer discovery if configured takes precedence over the receivers configured in the cluster databag. I have a `next` on line 45 which should skip the receiver configs if indexer discovery is configured.

Reply:
overlooked the `next` in there.  I think we need to document that behavior somewhere and in addition either fail the chef run or log a warning saying the receivers will be ignored if both are configured. I would prefer to fail the run.

Reply:
Talked to @gravesb about this. Personally, I don't think there is a reason to fail the run here because having the receivers configured is not going to cause any issues. indexer discovery will take precedence and everything should work as expected. I believe logging a warning is sufficient here and that way it will be consistent with how the cookbook handles similar cases. For example, when `receivers` attribute is not configured, we are only logging a warning. https://github.com/cerner/cerner_splunk/blob/stable/libraries/outputs.rb#L33.

For now, I made the update so that it logs a warning. I have also made the README update to mention that receivers configured in the data bag will be ignored if indexer discovery is enabled. 24053d683e023aa69c5b514f64a8b6cc6af294f2